docs/v1/api/iframe.md

```diff
@@ -0,0 +1,241 @@
+  }
+})
+```
+
+### getUrl
+
+<!-- eslint-skip -->
+
+```typescript
+getUrl(): Promise<string>;
+```
+
+Returns the iFrame URL.
+
+#### Returns
+
+- A promise that resolves to the iFrame URL
+
+#### Example
+
+```typescript
+const url = await iframe.getUrl()
+console.log(url) // "https://example.com/embed/video"
+```
+
+### on
+
+<!-- eslint-skip -->
+
+```typescript
+on<K extends keyof IFrameEvents>(eventName: K, listener: (...args: IFrameEvents[K]) => Awaitable<void>): void;
+```
+
+Subscribes to events emitted by the extension.
+
+#### Parameters
+
+- `eventName`: The name of the event to subscribe to
+- `listener`: The callback function for the event
+
+#### Example
+
+```typescript
+iframe.on('UpdateData', () => {
+  // Send updated data to the presence script
+  iframe.send({
+    video: {
+      currentTime: document.querySelector('video').currentTime,
+      paused: document.querySelector('video').paused
+    }
+  })
+})
+```
+
+## Events
+
+### UpdateData
+
+Emitted on every tick, used to update the data sent from the iFrame.
+
+#### Example
+
+```typescript
+iframe.on('UpdateData', () => {
+  // Send updated data to the presence script
+})
+```
+
+## Complete Example
+
+Here's a complete example of how to use the iFrame class:
+
+### iframe.ts
+
+```typescript
+const iframe = new iFrame()
+
+iframe.on('UpdateData', () => {
+  const video = document.querySelector('video')
+
+  if (video) {
+    iframe.send({
+      video: {
+        title: document.querySelector('.video-title')?.textContent,
+        currentTime: video.currentTime,
+        duration: video.duration,
+        paused: video.paused
+      }
+    })
+  }
+})
+```
+
+### presence.ts
+
+```typescript
+const presence = new Presence({
+  clientId: '123456789012345678'
+})
+
+// Store iFrame data
+let iFrameData: {
+  video?: {
+    title?: string
+    currentTime?: number
+    duration?: number
+    paused?: boolean
+  }
+} = {}
+
+// Listen for iFrame data
+presence.on('iFrameData', (data) => {
+  iFrameData = data
+})
+
+presence.on('UpdateData', async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: 'https://example.com/logo.png'
+  }
+
+  // Check if we have video data from the iFrame
+  if (iFrameData.video) {
+    const { title, currentTime, duration, paused } = iFrameData.video
+
+    presenceData.details = title || 'Watching a video'
+
+    if (paused) {
+      presenceData.state = 'Paused'
+      delete presenceData.startTimestamp
+      delete presenceData.endTimestamp
+    }
+    else {
+      presenceData.state = 'Playing'
+
+      // Calculate timestamps if we have currentTime and duration
+      if (currentTime && duration) {
+        const timestamps = getTimestamps(currentTime, duration)
+        presenceData.startTimestamp = timestamps[0]
+        presenceData.endTimestamp = timestamps[1]
+      }
+    }
+  }
+  else {
+    presenceData.details = 'Browsing'
+    presenceData.startTimestamp = Date.now()
+  }
```

Comment:
Tell them to setup a browsing timestamp outside of the updatedata, otherwise the time keeps resetting.